actionpack/lib/action_controller/test_case.rb

```diff
@@ -593,7 +607,14 @@ def process(action, http_method = 'GET', *args)
           @request.env['RAW_POST_DATA'] = args.shift
         end
 
-        parameters, session, flash = args
+        if args[0] && args[0].keys.any? { |k| REQUEST_KWARGS.include?(k) }
+          parameters = args[0][:params]
+          session =    args[0][:session]
+          flash =      args[0][:flash]
+        else
+          parameters, session, flash = args
```

Comment:
Since we are deprecating all documentation and tests should use the kwarg implementation and we can remove the old implementation from the documentation keeping only a few tests to make sure it works and it is deprecated.